veto/veto.cxx

```diff
@@ -1,1205 +1,1151 @@
-    Double_t zStartDecayVol = fTub1z-fTub1length-f_InnerSupportThickness;
-    Double_t zStartMagVol = fTub3z+fTub3length-f_InnerSupportThickness; //? is this needed, -f_InnerSupportThickness
-
-      Double_t d = f_VetoThickness+2*f_RibThickness+f_OuterSupportThickness;
-      Double_t slopex = (2 * m + d) / (fTub6z - fTub6length - zFocusX);
-      Double_t slopey = (fBtube + d) /(fTub6z-fTub6length - zFocusY);
-      Double_t zpos = fTub1z -fTub1length -f_LidThickness;
-      Double_t dx1 = slopex*(zpos - zFocusX);
-      Double_t dy  = slopey*(zpos - zFocusY);
-   // make the entrance window
-      //entrance lid
-      double wallThick = 20 * mm;  // wall thickness
-      TGeoVolume* T1Lid = MakeLidSegments(1, VetoStartInnerX / 2. + wallThick, VetoStartInnerY / 2. + wallThick);
-
-      tDecayVol->AddNode(T1Lid, 1, new TGeoTranslation(0, 0, zpos - zStartDecayVol+f_LidThickness/2.1));
-
-      //without segment1, recalculate the z and (half)length of segment 2:
-      //Take into account to remove the between seg1 and seg2 due to straw-veto station.
-      //and add this gap to the total length.
-      Double_t tgap=fTub2z-fTub1z-fTub2length-fTub1length;
-      fTub2z=fTub1z+fTub2length+tgap/2.;
-      fTub2length=fTub2length+fTub1length+tgap/2.;
-      TGeoVolume* seg2 = MakeSegments(fTub2length,dx1,dy,slopex,slopey,floorHeightA);
-      tDecayVol->AddNode(seg2, 1, new TGeoTranslation(0, 0, fTub2z-zStartDecayVol));
-
-   // make the exit window
-      Double_t dx2 = slopex*(fTub6z +fTub6length - zFocusX);
-      TGeoVolume *T6Lid = gGeoManager->MakeBox("T6Lid",supportMedOut,dx2,dy,f_LidThickness/2.);
-      T6Lid->SetLineColor(18);
-      T6Lid->SetLineColor(2);
-      T6Lid->SetFillColor(2);
-      tMaGVol->AddNode(T6Lid, 1, new TGeoTranslation(0, 0,fTub6z+fTub6length+f_LidThickness/2.+0.1*cm - zStartMagVol));
-      //finisMakeSegments assembly and position
-      top->AddNode(tDecayVol, 1, new TGeoTranslation(0, 0,zStartDecayVol));
-      top->AddNode(tMaGVol, 1, new TGeoTranslation(0, 0,zStartMagVol));
-
-
-// only for fastMuon simulation, otherwise output becomes too big
-     if (fFastMuon && fFollowMuon){
-        const char* Vol  = "TGeoVolume";
-        const char* Mag  = "Mag";
+
+    vetoMed = gGeoManager->GetMedium(vetoMed_name);   //! medium of veto counter, liquid or plastic scintillator
+    supportMedIn = gGeoManager->GetMedium(supportMedIn_name);       //! medium of support structure, iron, balloon
+    supportMedOut = gGeoManager->GetMedium(supportMedOut_name);     //! medium of support structure, aluminium, balloon
+    decayVolumeMed = gGeoManager->GetMedium(decayVolumeMed_name);   // decay volume, air/helium/vacuum
+
+    TGeoVolume* tDecayVol = new TGeoVolumeAssembly("DecayVolume");
+
+    // TGeoVolume* T1Lid = MakeLidSegments(1, VetoStartInnerX / 2. + wallThick, VetoStartInnerY / 2. + wallThick);
+    // tDecayVol->AddNode(T1Lid, 1, new TGeoTranslation(0, 0, zpos - zStartDecayVol+f_LidThickness/2.1));
```

Comment:
Is there a reason to keep these lines around?

Reply:
Well the code is technically obsolete, but I kept it just incase we plan to place anything in the future. Maybe I take it out then?

Reply:
Probably a new lid would be implemented differently anyway.
I think we can safely remove this. We can always look at older commits, if necessary.

Reply:
ok will do.